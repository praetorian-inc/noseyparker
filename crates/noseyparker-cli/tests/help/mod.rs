//! Tests for Nosey Parker `help` functionality

use super::*;

#[test]
fn no_args() {
    assert_cmd_snapshot!(noseyparker!().assert().failure());
}

#[test]
fn help() {
    assert_cmd_snapshot!(noseyparker_success!("help"));
}

#[test]
fn help_short() {
    assert_cmd_snapshot!(noseyparker_success!("-h"));
}

#[test]
fn help_scan() {
    with_settings!({
        filters => vec![
            (r"(?m)(scanning threads\s+)\[default: \d+\]", r"$1[default: DEFAULT]"),
        ],
    }, {
        assert_cmd_snapshot!(noseyparker_success!("help", "scan"));
    });
}

#[test]
fn help_scan_short() {
    with_settings!({
        filters => vec![
            (r"(?m)(scanning threads\s+)\[default: \d+\]", r"$1[default: DEFAULT]"),
        ],
    }, {
        assert_cmd_snapshot!(noseyparker_success!("scan", "-h"));
    });
}

#[test]
fn help_summarize() {
    assert_cmd_snapshot!(noseyparker_success!("help", "summarize"));
}

#[test]
fn help_summarize_short() {
    assert_cmd_snapshot!(noseyparker_success!("summarize", "-h"));
}

#[test]
fn help_report() {
    with_settings!({
        filters => vec![
            (r"(?m)(denoising threads when using the CPU\s+)\[default: \d+\]", r"$1[default: DEFAULT]"),
        ],
    }, {
        assert_cmd_snapshot!(noseyparker_success!("help", "report"));
    });
}

#[test]
fn help_report_short() {
    with_settings!({
        filters => vec![
            (r"(?m)(denoising threads when using the CPU\s+)\[default: \d+\]", r"$1[default: DEFAULT]"),
        ],
    }, {
        assert_cmd_snapshot!(noseyparker_success!("report", "-h"));
    });
}

#[test]
fn help_datastore() {
    assert_cmd_snapshot!(noseyparker_success!("help", "datastore"));
}

#[test]
fn help_rules() {
    assert_cmd_snapshot!(noseyparker_success!("help", "rules"));
}

#[test]
fn help_github() {
    assert_cmd_snapshot!(noseyparker_success!("help", "github"));
}

#[test]
fn help_github_short() {
    assert_cmd_snapshot!(noseyparker_success!("github", "-h"));
}

#[test]
fn help_github_repos() {
    assert_cmd_snapshot!(noseyparker_success!("help", "github", "repos"));
}

#[test]
fn help_github_repos_short() {
    assert_cmd_snapshot!(noseyparker_success!("github", "repos", "-h"));
}

// #[test]
// fn version_short() {
//     assert_cmd_snapshot!(noseyparker_success!("-V"));
// }

#[test]
fn version_long() {
    with_settings!({
        filters => vec![
            (r"(?m)^(    [^:]+:[ \t]+).*$", r"$1<PLACEHOLDER>")
        ],
    }, {
        assert_cmd_snapshot!(noseyparker_success!("--version"));
    });
}

#[test]
fn version_command() {
    // there is no `version` command
    assert_cmd_snapshot!(noseyparker_failure!("version"));
}
