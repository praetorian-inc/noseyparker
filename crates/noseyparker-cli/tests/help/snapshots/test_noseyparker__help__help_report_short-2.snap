---
source: crates/noseyparker-cli/tests/help/mod.rs
expression: stdout
---
Report detailed scan findings

Usage: noseyparker report [OPTIONS]

Options:
  -d, --datastore <PATH>  Use the specified datastore [env: NP_DATASTORE=] [default: datastore.np]
  -h, --help              Print help (see more with '--help')

Filtering Options:
      --max-matches <N>          Limit the number of matches per finding to at most N [default: 3]
      --min-score <SCORE>        Only report findings that have a mean score of at least N [default:
                                 0.05]
      --finding-status <STATUS>  Include only findings with the assigned status [possible values:
                                 accept, reject, mixed, null]

Output Options:
  -o, --output <PATH>    Write output to the specified path
  -f, --format <FORMAT>  Write output in the specified format [default: human] [possible values:
                         human, json, jsonl, sarif]

Global Options:
  -v, --verbose...       Enable verbose output
  -q, --quiet            Suppress non-error feedback messages
      --color <MODE>     Enable or disable colored output [default: auto] [possible values: auto,
                         never, always]
      --progress <MODE>  Enable or disable progress bars [default: auto] [possible values: auto,
                         never, always]
      --ignore-certs     Ignore validation of TLS certificates
